### PR TITLE
CMS Under Construction page

### PR DESCRIPTION
**Pull request by Ciuffi**
I wanted to add a page to the CMS that we can trigger through firebase in order to shut down the CMS.

The database update PR is here: https://github.com/nwplus/CMS-V2/pull/42

You can turn the CMS on/off through firebase (root/InternalWebsites/CMS/CMSOff)

### CODE REVIEW

**Comment by Ciuffi** [comment]
Here is the page:

image

**Comment by Ciuffi** [comment]
Here is what the page looks like:

image

File: components/CMSUnderConstructionPage.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  font-weight: bold;
+`;
+
+export default function CMSUnderConstructionPage({ date }) {
+  return (
+    <Container>
+      <MainContent>
+        <NwplusImage src={nwplusLogo} />
+        <BoldedTextLine>nwPlus CMS</BoldedTextLine>
+        <TextLine> The CMS is under maintainance. </TextLine>
```
Comment (acchiang) [inline components/CMSUnderConstructionPage.js:47]: nitpick: please change to "maintenance" with a E

File: components/CMSUnderConstructionPage.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        <TextLine>
+          {date
+            ? `It will be back up at ${formatDate(date.seconds)}`
+            : 'It will be back up soon!'}
+        </TextLine>
```
Comment (acchiang) [inline components/CMSUnderConstructionPage.js:52]: very robust wow i like

File: components/CMSUnderConstructionPage.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import styled from 'styled-components';
+import { formatDate } from '../utility/firebase';
+import nwplusLogo from '../assets/nwplus.svg';
+
+const NwplusImage = styled.img`
```
Comment (acchiang) [inline components/CMSUnderConstructionPage.js:5]: hahaha we can't be upset at non-club members calling us NwPlus or Nwplus or nwplUS anymore

File: utility/auth.js [hunk 4]
```diff
@@ -56,16 +37,7 @@ const Auth = ({ children }) => {
...
-        <LoadingScreenContainer>
-          <div>
-            <LoadingImage src={nwPlusReversed} />
-            <LoadingDiv>Authenticating...</LoadingDiv>
-          </div>
-        </LoadingScreenContainer>
-      ) : (
-        children
-      )}
+      {isLoading ? <LoadingPage /> : children}
```
Comment (acchiang) [inline utility/auth.js:40]: very nice

**Review by acchiang — APPROVED** [review]
lgtm but u should probably get a real dev to review too

File: utility/firebase.js [hunk 5]
```diff
@@ -298,3 +300,14 @@ export const updateFlags = async (id, flags) => {
   };
   return db.collection(Hackathons).doc(id).update(doc);
 };
+
+export const SubscribeToCMSStatus = (statuscb, datecb) => {
```
Comment (ianmah) [inline utility/firebase.js:304]:
```suggestion
export const subscribeToCMSStatus = (statuscb, datecb) => {
```
nit

**Comment by Ciuffi** [comment]
on hold till we figure out how we want to make the DB schema.

File: components/CMSUnderConstructionPage.js (hunk 1 above)
Comment (Ciuffi) [reply components/CMSUnderConstructionPage.js:47]: done!

File: components/CMSUnderConstructionPage.js (hunk 3 above)
Comment (Ciuffi) [reply components/CMSUnderConstructionPage.js:5]: it has to be capital because its a styled component :(

**Comment by Ciuffi** [comment]
@ianmah okie please re review now

File: components/CMSUnderConstructionPage.js (hunk 3 above)
Comment (ianmah) [reply components/CMSUnderConstructionPage.js:5]: NwPlus at leeeast

File: utility/CMSControl.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react';
+import { subscribeToCMSStatus } from './firebase';
+import CMSUnderConstructionPage from '../components/CMSUnderConstructionPage';
+import LoadingPage from '../components/LoadingPage';
+
+export default function CMSControl({ children }) {
+  const [CMSOffDate, setCMSOffDate] = useState(null);
```
Comment (ianmah) [inline utility/CMSControl.js:7]: not a blocker but would prefer cmsOffDate

File: components/LoadingPage.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  text-align: center;
+`;
+
+const LoadingImage = styled.img`
+  height: 100px;
+  width: 100px;
+  margin: auto;
+`;
+
+const LoadingDiv = styled.div`
```
Comment (ianmah) [inline components/LoadingPage.js:18]: not a blocker but why no `p` or text element? Also prefer label `Container` over `Div` for consistency

**Review by ianmah — APPROVED** [review]

File: components/CMSUnderConstructionPage.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+`;
+
+const MainContent = styled.div`
+  width: 400px;
+  height: 400px;
+  font-family: 'HK Grotesk';
+  font-size: 24px;
+  text-align: center;
+
+  position: absolute;
```
Comment (ianmah) [inline components/CMSUnderConstructionPage.js:22]: i enjoy flexbox but this is okay too

Comment (Ciuffi) [reply components/CMSUnderConstructionPage.js:22]: How would you do it with flexbox out of curiosity?

Comment (ianmah) [reply components/CMSUnderConstructionPage.js:22]: justify-content and align-content i think

**Pull request merged**